testng-core/src/test/kotlin/test/SimpleBaseTest.kt

```diff
@@ -0,0 +1,407 @@
+package test
+
+import org.assertj.core.api.Assertions.assertThat
+import org.testng.*
+import org.testng.annotations.ITestAnnotation
+import org.testng.internal.annotations.AnnotationHelper
+import org.testng.internal.annotations.DefaultAnnotationTransformer
+import org.testng.internal.annotations.IAnnotationFinder
+import org.testng.internal.annotations.JDK15AnnotationFinder
+import org.testng.xml.*
+import org.testng.xml.internal.Parser
+import java.io.*
+import java.nio.file.FileVisitResult
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.attribute.BasicFileAttributes
+import java.util.*
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.regex.Pattern
+import java.util.stream.Collectors
+
+open class SimpleBaseTest {
+
+    companion object {
+        private const val TEST_RESOURCES_DIR = "test.resources.dir"
+
+        @JvmStatic
+        fun run(vararg testClasses: Class<*>) = run(false, *testClasses)
+
+        @JvmStatic
+        private fun run(skipConfiguration: Boolean, tng: TestNG) =
```

Comment:
Ok to keep the function for compatibility purpose but I think `private fun TestNG.run(skipConfiguration: Boolean)` is "better" Kotlin.

Reply:
Yes. Agreed. Will get to it when the Kotlin migration is complete because either ways a round of refactoring of the code will be required to leverage more of the Kotlin syntax. But I think for now its going to be as good as premature optimisation because we still have Java tests that need to be moved over. It wouldnt matter even if the tests are migrated first because they will still use the same syntax of the Java base class which when refactored will start affecting all the tests. 

To me its the same effort irrespective of which way we migrate things from. Since I started with the base class, I would request you to please help me proceed with the approach (given the fact that nothing is breaking and this is only incremental refactoring)